File: mgyoutube-multitech/api-node/src/modules/usermodule.ts
import { UserCredential } from '../apimodel/usercredential';
import { User } from '../apimodel/user';
import { UserDataRepo } from '../repos/userdatarepo';


export interface UserModule {

    setUserDataRepo(udr: UserDataRepo): void;

    authUser(userCredential: UserCredential): Promise<User | undefined>;

    getUser(username: string): Promise<User | undefined>;

    createUpdateUser(user: User): Promise<User | undefined>
    createUser(user: User): Promise<User | undefined>;
    updateUser(userId: string, user: User): Promise<User | undefined>;
    removeUser(username: string): Promise<User | undefined>;

    getChildrenForParent(parentUsername: string): Promise<Array<User>>;
    addUpdateChildToParent(parentUsername: string, childUser: User): Promise<User | undefined>;
}

File: mgyoutube-multitech/api-python/modules/UserModule.py
from abc import ABC, abstractmethod
from apimodel.UserCredential import UserCredential
from apimodel.User import User


class UserModule(ABC):
    @abstractmethod
    def authUser(self, userCredential: UserCredential) -> User:
        pass

    @abstractmethod
    def createUser(self, user: User) -> User:
        pass

    @abstractmethod
    def getUser(self, username: str) -> User:
        pass

    @abstractmethod
    def removeUser(self, username: str) -> User:
        pass

    @abstractmethod
    def updateUser(self, userId: str, user: User) -> User:
        pass

    @abstractmethod
    def getChildrenForParent(self, parentUsername: str) -> list:
        pass

    @abstractmethod
    def addUpdateChildToParent(self, parentUsername: str, childUser: User) -> User:
        pass
